test/types/community/changes_from_36.test-d.ts

```diff
@@ -0,0 +1,103 @@
+
+// MongoClientOptions
+const options: MongoClientOptions = {};
+// .readPreference no longer accepts boolean
+expectNotType<boolean>(options.readPreference);
+// .pkFactory cannot be an empty object
+expectNotType<{}>(options.pkFactory);
+// .checkServerIdentity cannot be `true`
+expectNotType<true>(options.checkServerIdentity);
+// .sslCA cannot be string[]
+expectNotType<string[]>(options.sslCA);
+// .sslCRL cannot be string[]
+expectNotType<string[]>(options.sslCRL);
+// .sslCert cannot be a Buffer
+expectNotType<Buffer>(options.sslCert);
+// .sslKey cannot be a Buffer
+expectNotType<Buffer>(options.sslKey);
+// .sslPass cannot be a Buffer
+expectNotType<Buffer>(options.sslPass);
+
+// Legacy option kept
+expectType<PropExists<MongoClientOptions, 'w'>>(true);
+// Removed options
+expectType<PropExists<MongoClientOptions, 'wtimeout'>>(false);
+expectType<PropExists<MongoClientOptions, 'j'>>(false);
+expectType<PropExists<MongoClientOptions, 'bufferMaxEntries'>>(false);
+expectType<PropExists<MongoClientOptions, 'poolSize'>>(false);
+expectType<PropExists<MongoClientOptions, 'socketOptions'>>(false);
+expectType<PropExists<MongoClientOptions, 'reconnectTries'>>(false);
+expectType<PropExists<MongoClientOptions, 'reconnectInterval'>>(false);
+expectType<PropExists<MongoClientOptions, 'compression'>>(false);
+expectType<PropExists<MongoClientOptions, 'poolSize'>>(false);
+expectType<PropExists<MongoClientOptions, 'minSize'>>(false);
+expectType<PropExists<MongoClientOptions, 'socketOptions'>>(false);
+expectType<PropExists<MongoClientOptions, 'reconnectTries'>>(false);
+expectType<PropExists<MongoClientOptions, 'reconnectInterval'>>(false);
+expectType<PropExists<MongoClientOptions, 'useNewUrlParser'>>(false);
+expectType<PropExists<MongoClientOptions, 'useUnifiedTopology'>>(false);
+
+expectType<string | undefined>(options.authSource);
+expectType<LoggerLevel | undefined>(options.loggerLevel);
+expectType<ReadPreferenceMode | ReadPreference | undefined>(options.readPreference);
+expectType<boolean | undefined>(options.promoteValues);
+expectType<number | undefined>(options.family);
+expectType<boolean | undefined>(options.ssl);
+expectType<boolean | undefined>(options.sslValidate);
+expectAssignable<((host: string, cert: PeerCertificate) => Error | undefined) | undefined>(
+  options.checkServerIdentity
+);
+
+// compression options have simpler specification:
+// old way: {compression: { compressors: ['zlib', 'snappy'] }}
+expectType<PropExists<MongoClientOptions, 'compression'>>(false);
+expectType<('none' | 'snappy' | 'zlib')[] | undefined>(options.compressors);
+
+// Removed cursor API
+const cursor = new MongoClient('').db().aggregate();
+expectType<PropExists<typeof cursor, 'maxScan'>>(false);
+expectType<PropExists<typeof cursor, 'setCursorOption'>>(false);
+expectType<PropExists<typeof cursor, 'setReadPreference'>>(false);
+expectType<PropExists<typeof cursor, 'snapshot'>>(false);
+
+// Doesn't return cursor of new type:
```

Comment:
I feel like we talked about this, but I don't remember if there was supposed to be a test here or something else

Reply:
There were tests added as a part of a seperate ticket to correct the types with cursor#map, the tests are in test/types/mongodb.test-d.ts